cmd/conformance-tester/pkg/runner/waiters.go

```diff
@@ -27,32 +28,34 @@ import (
 	ctypes "k8c.io/kubermatic/v2/cmd/conformance-tester/pkg/types"
 	"k8c.io/kubermatic/v2/cmd/conformance-tester/pkg/util"
 	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/util/wait"
 
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/util/sets"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func waitForControlPlane(ctx context.Context, log *zap.SugaredLogger, opts *ctypes.Options, clusterName string) (*kubermaticv1.Cluster, error) {
-	log.Debug("Waiting for control plane to become ready...")
 	started := time.Now()
 	namespacedClusterName := types.NamespacedName{Name: clusterName}
 
-	err := wait.Poll(3*time.Second, opts.ControlPlaneReadyWaitTimeout, func() (done bool, err error) {
+	log.Infow("Waiting for control plane to become ready...", "timeout", opts.ControlPlaneReadyWaitTimeout)
+
+	err := wait.PollLog(log, 5*time.Second, opts.ControlPlaneReadyWaitTimeout, func() (transient error, terminal error) {
 		newCluster := &kubermaticv1.Cluster{}
 
 		if err := opts.SeedClusterClient.Get(ctx, namespacedClusterName, newCluster); err != nil {
 			if apierrors.IsNotFound(err) {
-				return false, nil
+				return err, nil
 			}
 		}
 
 		// Check for this first, because otherwise we instantly return as the cluster-controller did not
 		// create any pods yet
 		if !newCluster.Status.ExtendedHealth.AllHealthy() {
-			return false, nil
+			return errors.New("cluster is not all healthy"), nil
```

Comment:
Why do we use `errors` instead of `fmt.Errorf`? I thought we wanted to get rid of `errors`.